### DISCUSSION TITLE
How can pass extra (custom) parameters in send_message?

### DISCUSSION THREAD

**Discussion by surinderdeveloper** [python-smpplib/python-smpplib#141] [Q&A]
SMPP provider needs 3 more parameters to send sms. I am passing in the `send_message` function along with other params but those are not working. Is there any other way to pass it?  I want to pass these params (`PE_ID`, `TELEMARKETER_ID`, `TEMPLATE_ID`).

```python
pdu = client.send_message(
                                source_addr_ton=smpplib.consts.SMPP_TON_INTL,
                                source_addr_npi=smpplib.consts.SMPP_NPI_ISDN,
                                source_addr=sender_object.sender_id,
                                dest_addr_ton=smpplib.consts.SMPP_TON_INTL,
                                dest_addr_npi=smpplib.consts.SMPP_NPI_ISDN,
                                destination_addr=str(number),
                                message_payload=part,
                                data_coding=data_coding,
                                esm_class=esm_class,
                                validity_period='000000080000000R',
                                registered_delivery=True,
                                PE_ID=XXXXXXXXXXXXXXXXXXX,
                                TELEMARKETER_ID=XXXXXXXXXXXXXXXXXXX,
                                TEMPLATE_ID='')
```

**Comment by surinderdeveloper** [comment]
I added these fields in const.py and command.py but when I add template_id empty string or None then it is not working.
how can I add a field as an integer in command.py with a length of 19 character

**Comment by eigenein** [comment]
Hi Surinder,

In the current implementation any extra parameters should be defined in [`SubmitSM`](https://github.com/python-smpplib/python-smpplib/blob/51a5375a95fa16fa84f0eae5d721e861cc291504/smpplib/command.py#L579) class, otherwise they indeed wouldn't work. If they are compliant with the SMPP specs, please feel free to submit a pull request.

> how can I add a field as an integer in command.py with a length of 19 character

Could you elaborate on this? E.g. which field and where the limitation does come from?
